contracts/Oxo.sol

```diff
@@ -0,0 +1,575 @@
+        // horizontal
+
+        if(field[0][0] == 1 && field[0][1] == 1 && field[0][2] == 1) player1Win();
+        if(field[0][0] == 2 && field[0][1] == 2 && field[0][2] == 2) player2Win();
+
+        if(field[1][0] == 1 && field[1][1] == 1 && field[1][2] == 1) player1Win();
+        if(field[1][0] == 2 && field[1][1] == 2 && field[1][2] == 2) player2Win();
+
+        if(field[2][0] == 1 && field[2][1] == 1 && field[2][2] == 1) player1Win();
+        if(field[2][0] == 2 && field[2][1] == 2 && field[2][2] == 2) player2Win();
+
+        // vertical
+
+        if(field[0][0] == 1 && field[1][0] == 1 && field[2][0] == 1) player1Win();
+        if(field[0][0] == 2 && field[1][0] == 2 && field[2][0] == 2) player2Win();
+
+        if(field[0][1] == 1 && field[1][1] == 1 && field[2][1] == 1) player1Win();
+        if(field[0][1] == 2 && field[1][1] == 2 && field[2][1] == 2) player2Win();
+
+        if(field[0][2] == 1 && field[1][2] == 1 && field[2][2] == 1) player1Win();
+        if(field[0][2] == 2 && field[1][2] == 2 && field[2][2] == 2) player2Win();
+
+        // diagonal
+
+        if(field[0][0] == 1 && field[1][1] == 1 && field[2][2] == 1) player1Win();
+        if(field[0][0] == 2 && field[1][1] == 2 && field[2][2] == 2) player2Win();
+
+        if(field[2][0] == 1 && field[1][1] == 1 && field[0][2] == 1) player1Win();
+        if(field[2][0] == 2 && field[1][1] == 2 && field[0][2] == 2) player2Win();
+
+        // dead heat
+
+        if(_gameOn == 9) deadHeat();
+
+//        Debug('checkWiner end');
+    }
+
+    function stopChallenge()public{
+        // require(msg.sender==owner || msg.sender==player1Addr);
+
+        // Any can stop
+        require(now > timeStart1 + 10 seconds); //10 seconds \ 3 minutes
+        require(playerNum == 2);
+        player1Addr.transfer(bet);
+        bet = 0;
+        gameOn = false;
+        playerNum = 1;
+
+
+        StopChall(player1Addr, players[msg.sender], bet, "bet returned" );
+    }
+
+    function stopWaitingMove() public {
+        require(playerNum == 3);
+
+        if (movesCount == 1){
+            require(now > timeStart2 + 10 seconds);  // 3 minutes
+            player2Win();
+        }
+
+        if (movesCount % 2 == 0){
+            require(now > moveTime1 + 10 seconds);
+            player1Win();
+        }
+
+        if (movesCount % 2 != 0 && movesCount != 1 ){
+            require(now > moveTime2 + 10 seconds);
+            player2Win();
+        }
+
+    }
+
+    function deadHeat() internal {
+        player1Addr.transfer(bet);
+        player2Addr.transfer(bet);
+        DeadHeat(bet/1 ether, player1Addr, player2Addr, "bets returned");
+        gameOver(9);
+    }
+
+    function player1Win() private {
+//        require(msg.sender == player1Addr);
+        uint gain = bet*2*90/100; //this.balance*90/100;
+        msg.sender.transfer(gain);
+        Win(msg.sender, players[msg.sender], gain/1 ether, "win");
+        gameOver(1);
+    }
+
+    function player2Win() private {
```

Comment:
functions player2Win() and player1Win() look similar. is it possible to make just one function?